libslide/src/bignum/sub.rs

```diff
@@ -0,0 +1,148 @@
+#![allow(clippy::suspicious_arithmetic_impl)]
+use crate::bignum::utils::is_smaller;
+use crate::bignum::Bignum;
+use std::convert::TryInto;
+use std::mem;
+use std::ops;
+
+impl ops::Sub for Bignum {
+    type Output = Bignum;
+
+    fn sub(self, other: Bignum) -> Bignum {
+        // make the decimal vectors the same size
+        // fill lhs and rhs vectors as well
+        let mut lhs_dec: Vec<u8>;
+        let mut res_dec: Vec<u8>;
```

Comment:
I think it would be cheaper to set up the vecs as `i8`s here to avoid casting in a loop. LLVM is probably smart enough to optimize it out anyway, but here we make it one step closer and the optimizer can probably vectorize the casts.